pkg/optimize/common.go

```diff
@@ -5,9 +5,72 @@ SPDX-License-Identifier: Apache-2.0
 
 package optimize
 
-import "github.com/np-guard/vpc-network-config-synthesis/pkg/ir"
+import (
+	"sort"
+
+	"github.com/np-guard/models/pkg/ds"
+	"github.com/np-guard/models/pkg/netp"
+	"github.com/np-guard/models/pkg/netset"
+
+	"github.com/np-guard/vpc-network-config-synthesis/pkg/ir"
+)
 
 type Optimizer interface {
 	// attempts to reduce number of SG/nACL rules
 	Optimize() (ir.Collection, error)
 }
+
+// each IPBlock is a single CIDR. The CIDRs are disjoint.
+func SortPartitionsByIPAddrs[T ds.Set[T]](p []ds.Pair[*netset.IPBlock, T]) []ds.Pair[*netset.IPBlock, T] {
+	cmp := func(i, j int) bool {
+		if p[i].Left.FirstIPAddress() == p[j].Left.FirstIPAddress() {
+			return p[i].Left.LastIPAddress() < p[j].Left.LastIPAddress()
+		}
+		return p[i].Left.FirstIPAddress() < p[j].Left.FirstIPAddress()
+	}
+	sort.Slice(p, cmp)
```

Comment:
`slices.SortFunc` is preferred. See comment [here](https://pkg.go.dev/sort#Slice)